### ISSUE TITLE
JavaScript/TypeScript SDK type issues

### ISSUE THREAD

**Issue by mderriey** [svix/svix-webhooks#1451]
_Originally posted on Slack at <https://svixcommunity.slack.com/archives/C022BJ2EBHV/p1727206631611579>_

### Version

svix npm package v1.34.0.

### Platform

N/A

### Description

We're using the svix npm package to interact with the API and we found a case where the TypeScript types are incorrect.

The `Svix.endpoint.patch` function takes an `EndpointPatch` argument typed as follows:

```ts
export class EndpointPatch {
  // Most properties omitted for brevity
  'channels'?: Array<string>;
  'filterTypes'?: Array<string>;
}
```

These two properties do not allow `null`, although the API does accept `null` values for these properties as per <https://api.svix.com/docs#tag/Endpoint/operation/v1.endpoint.patch>, meaning we've had to use an `as` type assertion to avoid a TypeScript error.

Any chance we can see if openapi-generator can correctly type those?
All properties in this type are [marked as nullable](https://github.com/svix/svix-webhooks/blob/main/openapi.json#L1266-L1338) so I suspect it's an openapi-generator config we may be missing.

Thanks in advance!

**Comment by tasn** [comment]
Thanks for reporting!

**Comment by tasn** [comment]
I think it may be fixed by https://github.com/svix/svix-webhooks/pull/1480

**Comment by svix-onelson** [comment]
> I think it may be fixed by [#1480](https://github.com/svix/svix-webhooks/pull/1480)

Looks like it should be, yes.

The new generator produces the following:

```typescript
export class EndpointPatch {
    'channels'?: Array<string> | null;
    'filterTypes'?: Array<string> | null;
```

The updated models should be included in the next release.
